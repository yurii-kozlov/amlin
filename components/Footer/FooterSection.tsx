import React, { ReactElement, useState } from 'react';
import Image from 'next/image';
import { footerBlock } from 'types/footerBlock';
import Link from 'next/link';
import cn from 'classnames';
import arrowTop from 'images/icons/arrowTop.svg';
import arrowDown from 'images/icons/arrowDown.svg';
import styles from 'styles/layout/Footer/FooterSection.module.scss';

type FooterSectionProps = {
  section: footerBlock
}

export const FooterSection:React.FC<FooterSectionProps> = ({ section }):ReactElement => {
  const { title, sections } = section
  const [areListsVisible, setAreListsVisible ] = useState<boolean>(false);

  const showList = (): void => setAreListsVisible(!areListsVisible);

  return (
    <div className={styles.block}>
      <a
        className={styles.openingLink}
        href="#showList"
        onClick={showList}
        >
        <h3 className={styles.blockTitle}>{title}</h3>
        {areListsVisible ? (
          <Image
            alt="arrowTop"
            className={styles.arrow}
            src={arrowTop}
          />
        ): (
          <Image alt="arrowBottom" className={styles.arrow} src={arrowDown} />
        )}
      </a>
      <ul className={cn(styles.list, {[styles.listsVisible]: areListsVisible})} >
        {sections.map((sectionInfo) => (
          <li
            className={styles.listItem}
            key={sectionInfo.id}
            >
            <Link className={styles.link} href={sectionInfo.link}>
              {sectionInfo.name}
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
};
