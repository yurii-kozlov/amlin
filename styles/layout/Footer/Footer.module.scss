@import '../../utils/variables.scss';
@import '../../utils/mixins.scss';
@import '../../utils/placeholders.scss';

.section {
  padding-top: 3.5rem;

  @include onTablet {
    padding-top: 4rem;
  }

  @include onDesktop {
    padding-top: 5rem;
  }

  background-color: $colorBlackFooterBackground;
  @extend %smooth-transition;
}

.upperPart {
  display: flex;
  justify-content: space-between;
  margin-bottom: 2.6rem;

  @include onDesktop {
    margin-bottom: 4.5rem;
  }

  @include onMobile {
    display: block;
    text-align: center;
  }
}

.titles {
  @include onMobile {
    margin-bottom: 1.6rem;
  };
}

.titlesMain {
  font-style: normal;
  font-weight: 500;
  font-size: 1.8rem;
  line-height: 132.5%;
  color: $colorWhite;
  margin-bottom: 0.5rem;
  @include onDesktop {
    margin-bottom: 0.9rem;
  }
  @extend %smooth-transition;

  @include onDesktop {
    font-size: 3.8rem;
  }
}

.titlesSubtitle{
  color: $colorWhite;
  font-weight: 300;
  font-size: 1.2rem;
  line-height: 132.5%;
  font-style: normal;
  @extend %smooth-transition;

  @include onDesktop {
    font-size: 1.6rem;
  }
}

.input {
  min-width: 18rem;
  min-height: 4rem;

  @include onDesktop {
    min-width: 35rem;
    min-height: 5rem;
  }

  background-color: $colorBlack;
  border: 0.2rem solid $colorWhite;
  outline: none;
  border-radius: 0.4rem;
  margin-right: 2.3rem;
  @extend %smooth-transition;
  padding-left: 1rem;

  color: $colorWhite;
  font-style: normal;
  font-weight: 400;
  font-size: 1.1rem;

  @include onDesktop {
    font-size: 1.4rem;
  }

  line-height: 2rem;
  margin-bottom: 0.5rem;

}

.input:focus {
  border: 0.2rem solid $colorBlue3;
}

.input::placeholder {
  color: $colorWhite;
  font-style: normal;
  font-weight: 300;
  font-size: 1.1rem;
  line-height: 2rem;
  opacity: 0.6;

  @include onDesktop {
    font-size: 1.4rem;
  }
}

.subscriptionButton {
  @extend %buttonBlue;
  
}

.subscriptionButton:hover {
  background-color: $colorBlueHover;
}

.subscriptionButton:active {
  background-color: $colorBlueActive;
}

.error {
  color: $colorDanger;
  position: absolute;
  font-size: 1.1rem;

  @include onDesktop {
    font-size: 1.5rem;
  }
}

.inputBlock {
  display: inline-block;
  position: relative;
}

.middlePart {
  display: block;

  @include onDesktop {
    display: grid;
    grid-template-columns: repeat(5, 1fr);
    gap: 2rem;
    padding-bottom: 3.6rem;
    border-bottom: 0.1rem solid $colorWhite3;
  }
}

.lowerPart {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  padding-top: 2.7rem;
  padding-bottom: 3.5rem;

  @include onMobile {
    flex-wrap: wrap;
  }

  @include onTablet {
    padding-bottom: 4.5rem;
  }

  @include onDesktop {
    padding: 1.8rem 0;
  }
}

.copyright {
  font-style: normal;
  font-weight: 500;
  font-size: 1rem;
  line-height: 132.5%;
  color: $colorWhite;
  opacity: 0.6;

  @extend %smooth-transition;

  @include onDesktop {
    font-size: 1.2rem;
  }
}

.paymentSystemsLink:not(:last-child) {
  margin-right: 1rem;

  @include onDesktop {
    margin-right: 1.5rem;
  }
}

.socialNetworkLink:not(:last-child) {
  margin-right: 1rem;

  @include onDesktop {
    margin-right: 1.5rem;
  }
}

.socialNetworkLink {
  filter: brightness(50%);
}

.paymentSystems {
  @include onMobile {
    order: 1;
    margin: 1.5rem auto 0;
  }
}
