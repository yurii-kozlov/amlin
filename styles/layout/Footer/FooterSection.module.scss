.blockTitle {
  font-style: normal;
  font-weight: 700;
  font-size: 1.2rem;
  line-height: 100%;
  margin-bottom: 1.2rem;
  color: $colorWhite4;

  @extend %smooth-transition;

  @include onDesktop {
    font-size: 1.4rem;
    margin-bottom: 2.5rem;
    color: $colorWhite2;
  }
}

.block {
  position: relative;
  padding-bottom: 1.2rem;
  margin-bottom: 2rem;
  min-height: 3.6rem;
  @extend %smooth-transition;

  @include onDesktop {
    border-bottom: none;
    padding-bottom: 0;
    margin-bottom: 0;
  }
}

.block:last-child {
  margin-bottom: 0;
}

.block::after {
  content:'';
  display: inline-block;
  position: absolute;
  height: 0.1rem;
  width: 100%;
  background-color: $colorGrey1;

  @include onDesktop {
    display: none;
  }
}

.list {
  list-style: none;
  max-height: 0;
  overflow: hidden;
  padding-left: 1.3rem;
  transition: all 1s ease-in-out;
  
  @include onDesktop {
    padding-left: 0;
    overflow: visible;
    max-height: fit-content;
  }

  &Item {
    color: $colorWhite;
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 170%;
    margin-bottom: 0.3rem;

    @include onDesktop {
      font-size: 1.4rem;
      line-height: 100%;
      margin-bottom: 1rem;
    }
  }

  &Item:last-child {
    margin-bottom: 1rem;


    @include onDesktop {
      margin-bottom: 0;
    }
  }
}

.openingLink {
  @extend %smooth-transition;
  display: flex;
  justify-content: space-between;
  align-items: center;


  @include onDesktop {
    pointer-events: none;
    display: inline-block;
  }
}

.listsVisible {
  display: block;
  max-height: 30rem;
}

.openingLink:hover {
  opacity: 0.5;
}

.additionalInfo {
  color: $colorBlue4;
  @extend %smooth-transition;
}

.additionalInfo:hover {
  color: $colorBlue6;
}

.link {
  display: inline-block;
  @extend %smooth-transition;
}

.link:hover {
  color: $colorBlue5;
  transform: scale(1.05);
}

.arrow {
  display: inline-block;
  @extend %smooth-transition;

  @include onDesktop {
    display: none;
  }
}
