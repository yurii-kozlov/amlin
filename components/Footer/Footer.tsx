import React, { ReactElement, useRef, useState } from 'react';
import { footerBlocks } from 'api/footerBlocks';
import { useOnScreen } from 'hooks/useOnScreen';
import { FooterSection } from 'components/Footer/FooterSection';
import { paymentSystemsImages } from 'api/paymentSystemsImages';
import cn from 'classnames';
import { Container } from 'components/Container';
import faceBookIcon from 'images/icons/facebook.svg';
import instagramIcon from 'images/icons/instagram.svg';
import arrowTop from 'images/icons/arrowTop.svg';
import arrowDown from 'images/icons/arrowDown.svg';
import Image from 'next/image'
import styles from 'styles/layout/Footer/Footer.module.scss';
import footerSectionStyles from 'styles/layout/Footer/FooterSection.module.scss';

export const Footer: React.FC = (): ReactElement => {
  const [emailInput, setEmailInput] = useState<string>('');
  const [isEmailInputError, setIsEmailInputError] = useState<boolean>(false);
  const [isEmailInputEmptyError, setIsEmailInputEmtyError] = useState<boolean>(false);
  const [areListsVisible, setAreListsVisible ] = useState<boolean>(false);

  const handleEmailInput = (event: React.ChangeEvent<HTMLInputElement>): void => {
    if (!event.target.value) {
      setIsEmailInputEmtyError(true);
    } else {
      setIsEmailInputEmtyError(false);
    }

    if(event.target.value.length !==0 && !event.target.value.includes('@')) {
      setIsEmailInputError(true);
    } else {
      setIsEmailInputError(false);
    }

    setEmailInput(event.target.value);
  }

  const handleButtonClick = (): void => {
    if (emailInput.length === 0) {
      setIsEmailInputEmtyError(true);
    } else {
      setIsEmailInputEmtyError(false);
    }

    if(emailInput.length !==0 && !emailInput.includes('@')) {
      setIsEmailInputError(true);
    } else {
      setIsEmailInputError(false);
    }
  }

  const showList = (): void => setAreListsVisible(!areListsVisible);

  const ref: any = useRef<HTMLElement>();
  const onScreen: boolean = useOnScreen<HTMLElement>(ref, '150px');

  return (
    <section className={styles.section} data-testid="footer" ref={ref} >
      {onScreen && (
        <Container>
          <div className={styles.upperPart} >
            <div className={styles.titles}>
              <h1 className={styles.titlesMain} >Sign Up To Our Newsletter.</h1>
              <h2 className={styles.titlesSubtitle} >Be the first to hear about the latest offers.</h2>
            </div>
            <div className={styles.subscription}>
              <div className={styles.inputBlock}>
                <input
                  className={styles.input}
                  id="emailInput"
                  onChange={handleEmailInput}
                  placeholder='Your Email'
                  type="email"
                  value={emailInput}
                />
                {isEmailInputEmptyError && <p className={styles.error} >This field is required</p>}
                {isEmailInputError && <p className={styles.error} >Please enter a valid email address</p>}
              </div>

              <button
                className={styles.subscriptionButton}
                onClick={handleButtonClick}
                type="submit"
              >
                Subscribe
              </button>
            </div>
          </div>
          <div className={styles.middlePart}>
            {footerBlocks.map((block) => (
              <FooterSection
                key={block.id}
                section={block}
              />
            ))}
            <div className={footerSectionStyles.block}>
              <a
                className={footerSectionStyles.openingLink}
                href="#showList"
                onClick={showList}
              >
                <h3 className={footerSectionStyles.blockTitle}>Address</h3>
                {areListsVisible ? (
                  <Image alt="arrowTop" className={footerSectionStyles.arrow} src={arrowTop} />
                  ): (
                    <Image alt="arrowBottom" className={footerSectionStyles.arrow} src={arrowDown} />
                )}

              </a>
              <ul className={cn(footerSectionStyles.list, {[footerSectionStyles.listsVisible]: areListsVisible})}>
                <li className={footerSectionStyles.listItem}>1234 Street Adress City Address, 1234</li>
                <li className={footerSectionStyles.listItem}>Phones:
                  <a className={footerSectionStyles.contactLink} href="tel:(00) 1234 5678">
                    <span className={footerSectionStyles.additionalInfo}> (00) 1234 5678</span>
                  </a>
                </li>
                <li className={footerSectionStyles.listItem}>We are open: Monday-Thursday: 9:00 AM - 5:30 PM</li>
                <li className={footerSectionStyles.listItem}>Friday: 9:00 AM - 6:00 PM</li>
                <li className={footerSectionStyles.listItem}>Saturday: 11:00 AM - 5:00 PM</li>
                <li className={footerSectionStyles.listItem}>E-mail:
                  <a className={footerSectionStyles.contactLink} href="mailto:shop@email.com">
                    <span className={footerSectionStyles.additionalInfo}> shop@email.com</span>
                  </a>
                </li>
              </ul>
            </div>

          </div>
          <div className={styles.lowerPart}>
            <div className={styles.socialNetworks}>
              <a className={styles.socialNetworkLink} href="#facebook">
                <Image
                  alt='facebook'
                  height={15}
                  src={faceBookIcon}
                  width={15}
                />
              </a>
              <a className={styles.socialNetworkLink} href="#instagram">
                <Image
                  alt='instagram'
                  height={15}
                  src={instagramIcon}
                  width={15}
                />
              </a>
            </div>

            <div className={styles.paymentSystems}>
              {paymentSystemsImages.map((img) => (
                <a className={styles.paymentSystemsLink} href={`#${img.alt}`} key={`paymentSystem-${img.id}`}>
                  <Image alt={img.alt} className={styles.paymentSystemsIcon} src={img.imageLink} />
                </a>
                  ))}

            </div>

            <p className={styles.copyright} >Copyright Â© 2020 Shop Pty. Ltd.</p>
          </div>
        </Container>
      )}
    </section>
  );
}
