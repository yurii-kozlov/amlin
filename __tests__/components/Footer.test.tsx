import { render, screen } from '@testing-library/react';

import { Footer } from 'components/Footer/Footer';

describe('Footer', () => {
  test('Renders a header', () => {
    render(<Footer />);
    const headerElement = screen.getByText(/Sign Up To Our Newsletter./i)
    expect(headerElement).toMatchSnapshot();
  })
});
