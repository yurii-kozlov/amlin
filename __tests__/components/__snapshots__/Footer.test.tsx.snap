// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Renders a header 1`] = `
<h1
  class="titlesMain"
>
  Sign Up To Our Newsletter.
</h1>
`;
